From 40577992550c016cdebf9b8734d338c09381980e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Marchesin?= <marcheu@chromium.org>
Date: Fri, 16 Aug 2013 20:32:04 -0700
Subject: [PATCH] mesa: Workaround corruption on i915g.

The core of the issue here is that because of this new swizzle
we generate additional indirect texture accesses. So we will
need an optimization to remove those. In the meantime, this
workaround ensures that we don't generate additional indirect
texture accesses.

Change-Id: I762adb0b1fc9bc06e381670d1091d3990694056e
---
 src/mesa/state_tracker/st_glsl_to_tgsi.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/mesa/state_tracker/st_glsl_to_tgsi.cpp b/src/mesa/state_tracker/st_glsl_to_tgsi.cpp
index 64e0a8a..36bff75 100644
--- a/src/mesa/state_tracker/st_glsl_to_tgsi.cpp
+++ b/src/mesa/state_tracker/st_glsl_to_tgsi.cpp
@@ -2713,7 +2713,6 @@ glsl_to_tgsi_visitor::visit(ir_texture *ir)
        */
       coord = get_temp(glsl_type::vec4_type);
       coord_dst = st_dst_reg(coord);
-      coord_dst.writemask = (1 << ir->coordinate->type->vector_elements) - 1;
       emit(ir, TGSI_OPCODE_MOV, coord_dst, this->result);
    }
 
-- 
1.8.3

